R/btw_client.R

```diff
@@ -0,0 +1,64 @@
+#' Create an btw-enhanced ellmer chat client
+#'
+#' Creates an [ellmer::Chat] client, enhanced with the tools from
+#' [btw_register_tools()].
+#'
+#' @examples
+#' if (interactive()) {
+#'   withr::local_options(list(
+#'     btw.chat_client = ellmer::chat_ollama(model="llama3.1:8b")
+#'   ))
+#'
+#'   btw_chat <- btw_client()
+#'   chat$chat("How can I replace `stop()` calls with functions from the cli package?")
+#' }
+#'
+#' @param ... Additional arguments passed to [btw_register_tools()] to control
+#'   which tools are added to the chat client.
+#' @param client An [ellmer::Chat] client, defaults to [ellmer::chat_claude()].
+#'   You can use the `btw.chat_client` option to set a default client for new
+#'   `btw_client()` calls.
+#'
+#' @return Returns an [ellmer::Chat] object with additional tools registered by
+#'   [btw_register_tools()]. `btw_app()` returns the chat object invisibly, and
+#'   the chat object with the messages added during the chat session.
+#'
+#' @describeIn btw_client Create a btw-enhanced [ellmer::Chat] client
+#' @export
+btw_client <- function(..., client = NULL) {
+  if (is.null(client)) {
+    default <- getOption("btw.chat_client")
```

Comment:
The gander equivalent is `.gander_chat`, which we could follow here as `.btw_chat`, but I prefer `btw.chat` style options. Separately, I've been trying to nudge the Shiny team toward calling this chat object the `client`, because there we also have shinychat's interface and it gets a little confusing when they're both called `chat`.

So I'd like to go with `btw.chat_client`, with thanks to gander for showing how to do this so nicely :)

Reply:
Totally on board! I have some regrets on the gander option naming so this works for me😆